types/src/on_chain_config/execution_config.rs

```diff
@@ -106,7 +106,7 @@ pub struct ExecutionConfigV3 {
 impl Default for ExecutionConfigV3 {
```

Comment:
if we want to permanently have different things, I would have :
`default_for_genesis`
`default_if_missing`
basically `default_for_genesis`  is used for any new network. `default_if_missing`  is the backward compatible thing (needed for reply for example), to have old behavior if flag is missing

and I would remove Default macro completely, so people don't explicitly pick the one they want.

currently it is not critical, but if at any point in the future we add anything in the execution config that affects transaction output (not the order/which ones get on chain), we would need to have a separation, as replay will start with missing config